python/lsst/sims/alertsim/avro_utils.py

```diff
@@ -60,29 +61,25 @@ def catsim_to_avro(list_of_alert_dicts, session_dir, schemaURI='avsc/diasource.a
 
         serialization_timer = timer()
 
+
         #open file with a name of a chipNum in append mode
-        file_obj = open("json_output/"+session_dir+"/"+chipNum, 'a')
-        
-        avro_validated = False
+        with open(os.path.join("json_output/", session_dir, chipNum), 'a') as out_file:
 
-        for alert_dict in list_per_chip:
+            avro_validated = False
             
-            # json cannot serialize numpy types
-            #print("before scalarize")
-            #print(_print_types(alert_dict))
-            #_numpy_to_scalar(alert_dict)
-            #print("after scalarize")
-            #print(_print_types(alert_dict))
-
-            json.dump(alert_dict, file_obj)
+            for alert_dict in list_per_chip:
+            
+                json.dump(alert_dict, out_file)
 
-            if (ix==0 and avro_validated==False):
+                if (ix==0 and avro_validated==False):
             
-                writer.append(alert_dict)
-                print("Avro schema validated for this chunk")
-                avro_validated = True
+                    writer.append(alert_dict)
+                    
+                    print("Avro schema validated for this chunk")
+                    
+                    avro_validated = True
 
-        file_obj.close()
+        #file_obj.close()
         print("serialization for %d events on this chip took %s" % (len(list_per_chip), timer()-serialization_timer))
 
     writer.close()
```

Comment:
You should also be able to use `DataFileWriter` as a context, i.e.
```
with DataFileWriter(open("avsc/alert.avro", "wb"), DatumWriter(), alert_schema) as writer:
    ...code to write data...
```
then you won't need to remember to call `writer.close()`

Reply:
I'm a bit uncomfortable with so much nesting in that method

Reply:
Ok